Moya/Endpoint.swift

```diff
@@ -55,6 +55,12 @@ public class Endpoint<T> {
 
         return Endpoint(URL: URL, sampleResponse: sampleResponse, method: method, parameters: parameters, parameterEncoding: parameterEncoding, httpHeaderFields: newHTTPHeaderFields)
     }
+    
+    /// Convenience method for creating a new Endpoint with the same properties as the receiver, but with another parameter encoding.
+    public func endpointByAddingParameterEncoding(parameterEncoding: Moya.ParameterEncoding) -> Endpoint<T> {
+        
+        return Endpoint(URL: URL, sampleResponse: sampleResponse, method: method, parameters: parameters, parameterEncoding: parameterEncoding, httpHeaderFields: httpHeaderFields)
```

Comment:
Looks great! What do you think about changing the parameter to `newParameterEncoding` so we can avoid confusing it with the instance variable?